models/purchase.py

```diff
@@ -0,0 +1,28 @@
+from datetime import datetime, timedelta
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    active = fields.Boolean(default=True)
+    lifespan = fields.Integer(default=0)
+    usphone = fields.Char(string="Phone (US Format)")
+
+    def action_archive_purchase_orders(self):
+        for record in self:
+            if record.state == 'done' or record.state == 'cancel':
+                record.active = False
+            else:
+                raise UserError('Only allow archive the locked or canceled purchase orders')
```

Comment:
Also, you didn't make this method satisfy the condition `Only Purchase Manager can archive Purchase Order(s)` because later it can be called from elsewhere